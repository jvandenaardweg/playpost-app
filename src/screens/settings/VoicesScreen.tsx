import React from 'react';
import { NavigationScreenProp, NavigationRoute, NavigationStackScreenOptions } from 'react-navigation';
import { ScrollView } from 'react-native';
import { ButtonUpgrade } from '../../components/ButtonUpgrade';
import { VoiceSelectContainer } from '../../containers/VoiceSelectContainer';

interface Props {
  navigation: NavigationScreenProp<NavigationRoute>;
}

export class SettingsVoicesScreen extends React.PureComponent<Props> {
  static navigationOptions = ({ navigation }: { navigation: NavigationScreenProp<NavigationRoute> }): NavigationStackScreenOptions => {
    return {
      title: 'Voices',
      headerRight: <ButtonUpgrade onPress={navigation.getParam('handleOnPressUpgrade')} />,
    };
  }

  componentDidMount() {
    this.props.navigation.setParams({ handleOnPressUpgrade: this.handleOnPressUpgrade });
  }

  handleOnPressUpgrade = () => {
    this.props.navigation.navigate('Upgrade');
  }

  render() {
    return (
      <ScrollView>
        <VoiceSelectContainer />
      </ScrollView>
    );
  }
}
