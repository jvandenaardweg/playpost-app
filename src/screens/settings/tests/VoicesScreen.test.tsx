import React from 'react';
// tslint:disable-next-line: no-submodule-imports
import * as ShallowRenderer from 'react-test-renderer/shallow';

import { SettingsVoicesScreen } from '../VoicesScreen';

describe('SettingsVoicesScreen', () => {

  describe('rendering', () => {
    let wrapper: any;

    beforeEach(() => {
      const renderer = ShallowRenderer.createRenderer();
      renderer.render(<SettingsVoicesScreen />);
      wrapper = renderer.getRenderOutput();
    });

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
