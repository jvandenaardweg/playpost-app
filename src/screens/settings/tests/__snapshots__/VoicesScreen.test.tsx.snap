// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsVoicesScreen rendering should render correctly 1`] = `
<Memo>
  <Memo>
    <withNavigation(Connect(VoiceSelectContainerComponent)) />
  </Memo>
</Memo>
`;
